frontend/src/plugins/impl/DataTablePlugin.tsx

```diff
@@ -250,13 +257,27 @@ export const LoadingDataTableComponent = memo(
     }, [sorting, search, filters, searchQuery, props.fieldTypes, props.data]);
 
     // Column summaries
-    const { data: columnSummaries, error: columnSummariesError } =
-      useAsyncData(() => {
+    const { data: columnSummariesData, error: columnSummariesError } =
+      useAsyncData<ColumnSummaries>(() => {
         if (props.totalRows === 0) {
           return Promise.resolve({ summaries: [] });
         }
+        // Clear the summary first to avoid inconsistency with table
+        setSummaries({ summaries: [] });
         return props.get_column_summaries({});
-      }, [props.get_column_summaries, props.totalRows, props.data]);
+      }, [
+        props.get_column_summaries,
+        filters,
+        searchQuery,
```

Comment:
you are right - it looks like we were missing these deps so they were not re-rendering on filter changes. thanks for fixing